### ISSUE TITLE
beforestart event?

### ISSUE THREAD

**Issue by fxck** [blueimp/jQuery-File-Upload#2018]
To describe my problem, I want click the upload button, select files, open modal window, _create_ a record in the database and then start uploading and associate the files uploaded with the record I created(via say _id_).

The problem is that the start event is fired when the first upload begins, ending(at least the first file) before the end of my request creating record in the db resulting in me not being able to associate the record with the file being uploaded.

So how would I approach this? I could make my record creating request in _add_ event, but then I'd fire multiple times, I want all the files being uploaded in one "session" to be associated with just that one record I create.

I'm pretty much clueless, any ideas?

tl;dr
how do I 

click upload button -> select files -> make ajax request creating a record and an id -> put that id to formData -> have that same id in formData for all the files that I selected
